doc/instrumentation.rst

```diff
@@ -0,0 +1,120 @@
+***************
+Instrumentation
+***************
+
+In this section, we will explain how define and use instrumentation backends
+(such as ``bisect_ppx`` or ``landmarks``) so that you can enable and disable
+coverage via ``dune-workspace`` files or by passing a command-line flag or
+environment variable. In addition to providing an easy way to toggle
+instrumentation of your code, this setup avoids creating a hard dependency on
+the precise instrumentation backend in your project.
+
+Specifying what to instrument
+=============================
+
+When an instrumentation backend is activated, Dune will only instrument
+libraries and executables for which the user has requested instrumentation.
+
+To request instrumentation, one must add the following field to a library or
+executable stanza:
+
+.. code:: scheme
+
+   (library
+    (name ...)
+    (instrumentation
+     (backend <name>)))
+
+This field can be repeated multiple times in order to support various
+backends. For instance:
+
+.. code:: scheme
+
+   (library
+    (name foo)
+    (modules foo)
+    (instrumentation (backend bisect_ppx))
+    (instrumentation (backend landmarks)))
```

Comment:
Small bike shed: perhaps this syntax would be better:

```
(instrumentation
 (backend bisect_ppx)
 (backend landmarks))
```

Reply:
Agree. I implemented what was in the RFC, but this seems lighter. @jeremiedimino do you agree?

Reply:
The intent was in case we needed more options:

```
(instrumentation (backend bisect_ppx) (apply_to foo bar))
```

Which makes this field similar to `foreign_stubs`. If we don't plan to add options, I agree that the proposed is better.

My impression is that there might be options we will want to add, in particular the file selection mechanism since this came up in the bisect discussion. So I lean towards keeping the current syntax. Thoughts?

Reply:
Makes sense, I agree with the rationale.

Reply:
> The intent was in case we needed more options:

With a single `instrumentation` field, this is also possible:

```
(instrumentation
 (backend bisect_ppx (apply_to foo bar))
 (backend bar (apply_to baz bar)))
```

> So I lean towards keeping the current syntax. Thoughts?

I'm not a fan of it because it's inconsistent with how we handle other fields. Though i don't think it's a particularly big deal.

Reply:
Well, it's consistent with `foreign_stubs`. It seems that the general rule is: for lists where each element might require more parameters, we prefer a multi-field otherwise we prefer a single field taking a list.